src/auth/googleauth.ts

```diff
@@ -269,8 +269,10 @@ export class GoogleAuth {
           res && res.headers && res.headers['metadata-flavor'] === 'Google';
     } catch (e) {
       if ((e as NodeJS.ErrnoException).code !== 'ENOTFOUND') {
-        // Unexpected error occurred. TODO(ofrobots): retry if this was a
-        // transient error.
+        // Unexpected error occurred. Retry once.
+        if (!isRetry) {
+          return this._checkIsGCE(true);
```

Comment:
For correctness' sake I think this should be `return await this._cehckIsGCE(true)`.

Reply:
Yes, fixed!